cl/search/types.py

```diff
@@ -106,3 +107,6 @@ class EventTable(StrEnum):
     DOCKET_ENTRY = "search.DocketEntry"
     RECAP_DOCUMENT = "search.RECAPDocument"
     UNDEFINED = ""
+
+
+ESCursor = namedtuple("ESCursor", ["search_after", "reverse", "search_type"])
```

Comment:
is there a reason to use a `namedtuple` instead of a `dataclass`? `Dataclasses` provide the same data storage capabilities with the added benefit of built-in type hints and cleaner syntax. This can improve code readability and maintainability.

Reply:
Thanks for the suggestion. Using a `dataclass` can improve readability, so I've made the change it to:

```
@dataclass(frozen=True)
class ESCursor:
    search_after: AttrList | None
    reverse: bool
    search_type: str
```
The `frozen=True` to make it immutable as the `namedtuple`